examples/step-58/step-58.cc

```diff
@@ -0,0 +1,673 @@
+          {
+            for (unsigned int k = 0; k < dofs_per_cell; ++k)
+              {
+                for (unsigned int l = 0; l < dofs_per_cell; ++l)
+                  {
+                    const std::complex<double> i = {0, 1};
+
+                    cell_matrix_lhs(k, l) +=
+                      (-i * fe_values.shape_value(k, q_index) *
+                         fe_values.shape_value(l, q_index) +
+                       time_step / 4 * fe_values.shape_grad(k, q_index) *
+                         fe_values.shape_grad(l, q_index) +
+                       time_step / 2 * potential_values[q_index] *
+                         fe_values.shape_value(k, q_index) *
+                         fe_values.shape_value(l, q_index)) *
+                      fe_values.JxW(q_index);
+
+                    cell_matrix_rhs(k, l) +=
+                      (-i * fe_values.shape_value(k, q_index) *
+                         fe_values.shape_value(l, q_index) -
+                       time_step / 4 * fe_values.shape_grad(k, q_index) *
+                         fe_values.shape_grad(l, q_index) -
+                       time_step / 2 * potential_values[q_index] *
+                         fe_values.shape_value(k, q_index) *
+                         fe_values.shape_value(l, q_index)) *
+                      fe_values.JxW(q_index);
+                  }
+              }
+          }
+
+        cell->get_dof_indices(local_dof_indices);
+        constraints.distribute_local_to_global(cell_matrix_lhs,
+                                               local_dof_indices,
+                                               system_matrix);
+        constraints.distribute_local_to_global(cell_matrix_rhs,
+                                               local_dof_indices,
+                                               rhs_matrix);
+      }
+  }
+
+
+  // @sect4{Implementing the Strang splitting steps}
+
+  // Having set up all data structures above, we are now in a position to
+  // implement the partial steps that form the Strang splitting scheme. We
+  // start with the half-step to advance the phase, and that is used as the
+  // first and last part of each time step.
+  //
+  // To this end, recall that for the first half step, we needed to
+  // compute
+  // $\psi^{(n,1)} = e^{-i\kappa|\psi^{(n,0)}|^2 \tfrac
+  //  12\Delta t} \; \psi^{(n,0)}$. Here, $\psi^{(n,0)}=\psi^{(n)}$ and
+  //  $\psi^{(n,1)}$
+  // are functions of space and correspond to the output of the previous
+  // complete time step and the result of the first of the three part steps,
+  // respectively. A corresponding solution must be computed for the third
+  // of the part steps, i.e.
+  // $\psi^{(n,3)} = e^{-i\kappa|\psi^{(n,2)}|^2 \tfrac
+  //  12\Delta t} \; \psi^{(n,2)}$, where $\psi^{(n,3)}=\psi^{(n+1)}$ is
+  // the result of the time step as a whole, and its input $\psi^{(n,2)}$ is
+  // the result of the spatial step of the Strange splitting.
```

Comment:
`Strange splitting` -> `Strang splitting`

Reply:
@jppelteret you should not have made that suggestion - I would have loved to learn about that other type of splitting, the name sounded promising :rofl:

Reply:
@kronbichler Indeed, it was a missed opportunity to have some fun. But there's still time for you to claim ownership of it ;-) How would you choose to define Strange splitting?

Reply:
You totally all missed out on having a field day with my mistake!